### ISSUE TITLE
[Question] Any way to determine when something is used as part of email generation?

### ISSUE THREAD

**Issue by jconroy** [woothemes/woocommerce#9409]
Hi Guys,

Is there any way when hooking into the `woocommerce_order_item_meta_start`  and  `woocommerce_order_item_meta_end` actions or when filtering within ``display_item_downloads()` to determine whether they are called on the front end or when be used as part of email generation?

At the moment they are used within both the `email-order-items.php` template file https://github.com/woothemes/woocommerce/blob/master/templates/emails/email-order-items.php#L44-L58 and regular `order-details-items.php` https://github.com/woothemes/woocommerce/blob/master/templates/order/order-details-item.php#L34-L39 templates.

I just can't find a consistent way when using them to either only do something when used on the site or only when being used in the email template - so thought I'd reach out.

Thanks for the thinking in advance!

**Comment by WPprodigy** [comment]
I think you may be approaching this incorrectly. Those hooks are to do stuff before and after and links are displayed. 

What you need to do is actually alter `display_item_downloads()` - https://github.com/woothemes/woocommerce/blob/f6526d72a72e3e2b110ed6c0f7b18329239c6667/includes/abstracts/abstract-wc-order.php#L2119-L2139

Which uses `get_item_downloads()`, which has a filter in it: https://github.com/woothemes/woocommerce/blob/f6526d72a72e3e2b110ed6c0f7b18329239c6667/includes/abstracts/abstract-wc-order.php#L2116

So that it was you would want to use to alter the download links that are shown.

**Comment by jconroy** [comment]
@WPprodigy I understand that - that is precisely what I am doing - sorry I wasn't specific enough.

Let me rephrase - If we filter within `get_item_downloads()` (which is used by display_item_downloads())  is there any way to tell if we are filtering what is being used to display within the emails vs elsewhere?

**Comment by WPprodigy** [comment]
I haven't tried it, but should be possible. Try wrapping the filter in an account page conditional maybe: https://docs.woothemes.com/document/conditional-tags/

Those my be running to late though, so may need to find an earlier hook to hook your filter onto.

**Comment by franticpsyx** [comment]
@jconroy `did_action( 'woocommerce_email_before_order_table' ) > did_action( 'woocommerce_email_after_order_table' )`

**Comment by jconroy** [comment]
:+1: thanks @franticpsyx
